airflow/models/taskinstance.py

```diff
@@ -895,6 +895,19 @@ def refresh_from_db(self, session: Session = NEW_SESSION, lock_for_update: bool
         else:
             self.state = None
 
+    @classmethod
+    def get_by_key(cls, ti_key, session) -> TaskInstance:
+        return (
+            session.query(TaskInstance)
+            .filter(
+                TaskInstance.task_id == ti_key.task_id,
+                TaskInstance.dag_id == ti_key.dag_id,
+                TaskInstance.run_id == ti_key.run_id,
+                TaskInstance.map_index == ti_key.map_index,
+            )
+            .one_or_none()
+        )
```

Comment:
Don’t particularly like this. Putting all accessors on the model class should be considered an antipattern (search for how the active record pattern violates the single responsibility principle for details). I know Airflow already does this a lot, but let’s not pile on this.

Reply:
What would you propose?  Where should this logic go?  This exists so that `_render_filename` can accept either TI or TI Key.  Would you just inline it in _render_filename?

Reply:
Putting the logic in `_render_filename` seems fine to me. Or in a free function.

Reply:
don't need to reuse, but sticking it in (private) free function just for readability

Reply:
I know it has been a while but in which case `_render_filename` is passed a TI key? I was going through this PR against to use the `TaskContextLogger` in executors and I cannot understand why `_render_filename` needs to accept a TI key